proto/proto/requests.proto

```diff
@@ -31,14 +31,37 @@ message GetBlockHeaderByNumberRequest {
 }
 
 // State synchronization request.
+//
+// Specifies state updates the client is intersted in. The server will return
+// the first block which contains a note matching `note_tags`. And the
+// corresponding updates to `nullifiers` and `account_ids` for that block
```

Comment:
Very minor detail, but it might be worth mentioning here (on the request instead of `block_num`) that if there is no match, the chain tip is returned?

Reply:
updated